inc/namespace.php

```diff
@@ -456,3 +459,92 @@ function on_request_stats( TransferStats $stats ) {
 
 	on_aws_guzzle_request_stats( $stats );
 }
+
+/**
+ * Get the EC2 Instance metadata.
+ *
+ * This will cause a remote request to the metadata service when the cache is empty.
+ *
+ * @return array $array(
+ *    accountId: string,
+ *    architecture: string,
+ *    availabilityZone: string,
+ *    billingProducts: string,
+ *    devpayProductCodes: string,
+ *    marketplaceProductCodes: string,
+ *    imageId: string,
+ *    instanceId: string,
+ *    instanceType: string,
+ *    kernelId: string,
+ *    pendingTime: string,
+ *    privateIp: string,
+ *    ramdiskId: string,
+ *    region: string,
+ *    version: string,
+ * )
+ */
+function get_ec2_instance_metadata() : array {
+	$has_cache = false;
+	$cache_key = 'altis.ec2_instance_metadata';
+	$cached_data = apcu_fetch( $cache_key, $has_cache );
+	if ( $has_cache ) {
+		return $cached_data;
+	}
+
+	$client = new Client();
+
+	try {
+		$request = $client->request( 'GET', 'http://169.254.169.254/latest/dynamic/instance-identity/document', [
+			'timeout' => 1,
+			'on_stats' => __NAMESPACE__ . '\\on_request_stats',
+		] );
+	} catch ( Exception $e ) {
+		trigger_error( sprintf( 'Unable to get instance metadata. Error: %s', $e->getMessage() ), E_USER_NOTICE );
+		apcu_store( $cache_key, [] );
+		return [];
+	}
+
+	if ( $request->getStatusCode() !== '200' ) {
+		trigger_error( sprintf( 'Unable to get instance metadata. Returned response code: %s', $request->getStatusCode() ), E_USER_NOTICE );
+		apcu_store( $cache_key, [] );
+		return [];
+	}
+
+	$metadata = json_decode( $request->getBody(), true );
```

Comment:
Looks like you can just do `$request->json();` here

Reply:
Hmm atleast in the version of Guzzle we have, it looks like that isn't in `MessageInterface`

Reply:
Sad times 😢 As you were...